js/modules/timeline.src.js

```diff
@@ -0,0 +1,431 @@
+/**
+ * @license  @product.name@ JS v@product.version@ (@product.date@)
+ * Timeline series
+ *
+ * (c) 2010-2018 Highsoft AS
+ * Author: Daniel Studencki
+ *
+ * License: www.highcharts.com/license
+ */
+'use strict';
+import H from '../parts/Globals.js';
+
+var addEvent = H.addEvent,
+    each = H.each,
+    extend = H.extend,
+    merge = H.merge,
+    pick = H.pick,
+    Point = H.Point,
+    Series = H.Series,
+    seriesType = H.seriesType,
+    SVGElement = H.SVGElement,
+    wrap = H.wrap;
+
+/**
+ * Utils
+ */
+
+// Utility to check, whether the SVGElement is visible
+extend(SVGElement.prototype, {
+    isVisible: function () {
+        return this.visibility !== 'hidden' ? true : false;
+    }
+});
+
+// Within this wrap method is necessary to save the current animation params,
+// because the data label target position (after animation) is needed to align
+// connectors.
+wrap(H.SVGElement.prototype, 'animate', function (proceed, params) {
+    if (this.targetPosition) {
+        this.targetPosition = params;
+    }
+    return proceed.apply(this, Array.prototype.slice.call(arguments, 1));
+});
+
+/**
+ * Timeline Series definition.
+ */
+
+seriesType('timeline', 'line', {
+    showInLegend: false,
+    colorByPoint: true,
+    stickyTracking: false,
+    lineWidth: 0,
+    tooltip: {
+        headerFormat: '<span style="color:{point.color}">● </span>' +
+            '<span style="font-weight: bold;">{point.point.date}</span><br/>',
+        pointFormat: '{point.description}'
+    },
+    states: {
+        hover: {
+            lineWidthPlus: 5,
+            halo: {
+                size: 0
+            }
+        }
+    },
+    dataLabels: {
+        enabled: true,
+        allowOverlap: true,
+        backgroundColor: '#fff',
+        format: '<span style="color:{point.color}">● </span>' +
+            '<span style="font-weight: bold;>{point.date}</span><br/>' +
+            '{point.label}',
+        borderWidth: 1,
+        borderColor: 'gray',
+        distance: 100,
+        alternate: true,
+        verticalAlign: 'middle',
+        color: '#000'
+    },
+    marker: {
+        enabledThreshold: 0,
+        symbol: 'square',
+        radius: 10,
+        height: 15
+    }
+}, {
+    alignDataLabel: function () {
+        var series = this,
+            xAxis = series.xAxis,
+            xAxisLen = xAxis.len,
+            points = series.points,
+            point = arguments[0],
```

Comment:
You declare a function `alignDataLabel` and proceed on the same function `alignDataLabel` ?